internal/repository/repository.go

```diff
@@ -0,0 +1,110 @@
+package repository
```

Comment:
This is not the state. The state is in the storage package! This is a repository function thats why we named it repository...

Reply:
I've asked chat gpt about names suggestions but I didn't like anything of what it proposed.

I'm open to rename the folder `data`, but I would leave this for a future task.
Why? Because repository will get bigger and bigger, as it will require more retrieval and store methods. We were thinking about specific repositories, each per data type, so to better separate responsibilities. Example.
chainid.go:
```go
type chainID struct {}

func (c *chainId) GetChainID() {
	// ...
}
```
currentblock.go:
```go
type currentBlock struct {}

func (c *currentBlock) GetCurrentBlock() {
	// ...
}
```
service.go:
```go
type Service struct {
	*chainId
	*contract
	*currentBlock
	// ....
}
```

All the files would still be stored in the data/repository folder

Reply:
lets keep it as is and merge this because we are under pressure. But I am not 100% sure in this.